File: 2900-longest-unequal-adjacent-groups-subsequence-i/2900-longest-unequal-adjacent-groups-subsequence-i.cpp
class Solution {
public:
    vector<string> getLongestSubsequence(vector<string>& words, vector<int>& groups) {
        int n = words.size(), curr = -1;
        vector<string> ans;
        for (int i = 0; i < n; ++i){
            if (curr == -1 || curr != groups[i]){
                curr = groups[i];
                ans.push_back(words[i]);
            }
        }

        return ans;
    }
};

File: 2900-longest-unequal-adjacent-groups-subsequence-i/2900-longest-unequal-adjacent-groups-subsequence-i.py
class Solution:
    def getLongestSubsequence(self, words: List[str], groups: List[int]) -> List[str]:
        ans, n = [], len(words)
        curr = -1

        for i in range(n):
            if curr == -1 or curr != groups[i]:
                curr = groups[i]
                ans.append(words[i])

        return ans